kernel/src/fs/epoll/epoll_file.rs

```diff
@@ -527,7 +546,15 @@ impl EpollEntry {
     /// This method must be called while holding the lock of the ready list. This is the only way
     /// to ensure that the "is ready" state matches the fact that the entry is actually in the
     /// ready list.
-    pub fn set_ready(&self, _guard: &MutexGuard<VecDeque<Weak<EpollEntry>>>) {
+    pub fn set_ready_mutex(&self, _guard: &MutexGuard<LinkedList<Weak<EpollEntry>>>) {
+        self.is_ready.store(true, Ordering::Relaxed);
+    }
+
+    /// See `set_ready_mutex`.
+    pub fn set_ready_spinlock(
```

Comment:
I am splitting a ready list into two lists because the `pop_ready` function may hold the ready list for too long, preventing the interrupt callbacks from executing `push_ready`. I have tested this situation and the Nginx works well, so maybe we can simply change the ready list to `SpinLock<LinkedList<Weak<EpollEntry>>, LocalIrqDisabled>`.

Reply:
Yes, I understand your motivation. But my point is that using two locks can easily lead to race conditions. I don't think it's right, as explained above.

Reply:
Ok, I have removed the two locks implementation and only changed the ready list to `SpinLock`, which will also fix the problem.

Reply:
Well, while using `SpinLock` can be a hack to support Nginx, using `SpinLock` but the critical section can be arbitrarily long (note that the user program can control the number of files registered in the epoll) does not seem right in general. 

Also, the second problem is that the `inner` field in the `EpollEntry` is currently protected by a mutex, and the mutex is acquired while the ready lock is held. That's not correct, either.

Reply:
I am adding the Nginx benchmark based on #1331. The command `ab -n 10000 -c 1` works fine without this PR. However, the benchmark gets stuck even with this PR when the concurrency (`-c 1`) is set to a value greater than 1. 

As a result, I think we can close this PR and fix the problem when we have a better solution.

Reply:
Of course, we can definitely leave the job as future work by closing this PR since you said it doesn't block anything.